versions/development/SPEC.md

```diff
@@ -2602,9 +2602,22 @@ In this example, `i`, and `f` are inputs to this task even though `i` is not dir
 
 Workflows have inputs that must be satisfied to run them, just like tasks. Inputs to the workflow are provided as a key/value map where the key is of the form `workflow_name.input_name`.
 
-* If a workflow is to be used as a sub-workflow it must ensure that all of the inputs to its calls are satisfied.
-* If a workflow will only ever be submitted as a top-level workflow, it may optionally leave its tasks' inputs unsatisfied. This then forces the engine to additionally supply those inputs at run time. In this case, the inputs' names must be qualified in the inputs as `workflow_name.task_name.input_name`.
-
+Workflows call tasks (and subworkflows) that have inputs as well. These inputs can be filled
+in by the calling workflow, or left empty for the user to fill in. The rules for defining inputs are as follows:
+
+* Inputs should be fully qualified. For example `workflow_name.workflow_input`,
+  `workflow_name.some_task.task_input`, `workflow_name.subworkflow_name.subwf_task.input_name`
+  ,`workflow_name.subworkflow_name.another_subworkflow.yet_another_sub_wf.subwf_input` etc.
+  For more information checkout
+  [the section on fully qualified names](#fully-qualified-names--namespaced-identifiers).
+* *Optional* inputs (or required inputs with defaults) for  tasks are not required to be
+  satisfied.
+* It is good practice for workflows to provide the *required* inputs for its tasks. If this is
+  not done the workflow cannot be called as a subworkflow as the call `input:` section does
+  not allow namespaced inputs. Also, if a workflow does not fill its tasks' required options then these should be supplied by the user, while not being listed in the workflow `input` section. This can cause confusion and make the workflow harder to use.
+  Engines may optionally enforce the good behavior of supplying all tasks' required inputs by the
```

Comment:
Well, I made this optional on purpose, so I would not burden the engine developers with this. I can imagine it is not as straightforward as it may sound. But I am curious what the engine developers actually think. @mlin, @cjllanwarne ?